cmd/stackdriver-prometheus-sidecar/main.go

```diff
@@ -231,7 +232,10 @@ func main() {
 	a.Flag("web.listen-address", "Address to listen on for UI, API, and telemetry.").
 		Default("0.0.0.0:9091").StringVar(&cfg.listenAddress)
 
-	a.Flag("filter", "PromQL-style label matcher which must pass for a series to be forwarded to Stackdriver. May be repeated.").
```

Comment:
It looks like all the existing use-cases of `filter` can be covered by new implementation. I lean to keep using `filter` as this PR only expanded feature set of `filter` - which means it is backward compatible to existing use-cases.

Reply:
Unfortunately, `--filterset` is not backwards compatible with `--filter`, since multiple `--filter` flags are AND-ed together (all filters need to match for time series to be forwarded), while multiple `--filterset` flags are OR-ed (any filterset can match for time series to be forwarded).

#106 provides a bit more background and some discussion on handling this.